backend/src/bundles/auth/auth.service.ts

```diff
@@ -19,7 +16,7 @@ class AuthService {
     }
 
     private async verifyLoginCredentials(
-        userRequestDto: UserSignInRequestDto,
+        userRequestDto: UserAuthRequestDto,
     ): Promise<UserModel> {
         const user = (await this.userService.find({
             email: userRequestDto.email,
```

Comment:
I have a problem with `find` method inside `user.repository`. If I update the repo type to UserEntity I need to decide whether to return user with `passwordHash` or `userDetails`. Because in `verifyCredentials` I need hash. Would it be a good practice to return the combined fields?

Reply:
Oh, I'm sorry. I am inattentive, it's too late ahahaha

Reply:
But there is another issue you'll encounter is the inability to access the hash since it's private in `UserEntity`. To address this, you could create a public getter to retrieve it for now. I'll think about options how we can manage this better later on.

Reply:
Yes, I have already encountered this problem and thought how I can manage it better. Also, I need both password hash and user details inside verifyCredentials where I retrieve a user by email but we don't have a dedicated method for this. I hope I'll come with some ideas in the morning

Reply:
I'm not sure If I understood your issue correctly, but could be something like this 
`repository.find` return entire entity with all fields -> `service.find` also returns entity -> `verifyCredentials` retrieves hash with getter and returns entity -> `signIn` returns `{ user: userEntity.toObject(), token }`

Reply:
decided to refactor this one later, resolved for now